tests/websocket.test.js

```diff
@@ -0,0 +1,47 @@
+import WS from 'jest-websocket-mock';
+import { createWatcher } from '../src';
+import { calls, results } from './shared';
+
+describe('websocket', () => {
+  let server;
+
+  beforeAll(async () => {
+    server = new WS('ws://localhost:1234', { jsonProtocol: true });
+    server.on('connection', socket => {
+      socket.on('message', data => {
+        const json = JSON.parse(data);
+        socket.send(JSON.stringify({ jsonrpc: '2.0', id: json.id, result: results[json.id - 1] }));
+      });
+    });
+  });
+
+  test('requests using websocket endpoint', async () => {
+    const config = {
+      rpcUrl: 'ws://localhost:1234',
```

Comment:
at some point i wonder if it'd be worth spinning up a `makerdao/testchain` for some integration-style tests?

Reply:
yeah, that could be something worth doing.

Reply:
On second thought, given that Multicall is only ever reading blockchain state (and the result for a given request is therefore deterministic/always knowable and easily mockable) I wonder if using a local testchain for integration tests would be worthwhile?

cc @levity

Reply:
it depends on whether you think the actual websocket-level logic is worth testing. for example, with a websocket mock, you might not catch a bug where you pass in an address for a websocket connection but the address is ignored or overridden by some other code. with a real testchain on the other end, you can compare the response you get to the data you know is in the testchain, as opposed to a mock you set up.